src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotUseInsecureCryptographicAlgorithmsTests.cs

```diff
@@ -748,10 +2872,31 @@ protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
             return new CSharpDoNotUseInsecureCryptographicAlgorithmsAnalyzer();
         }
 
-        private const string CA5350RuleName = DoNotUseInsecureCryptographicAlgorithmsAnalyzer.DoNotUseBrokenCryptographicRuleId;
-        private const string CA5351RuleName = DoNotUseInsecureCryptographicAlgorithmsAnalyzer.DoNotUseWeakCryptographicRuleId;
-
-        private readonly string _doNotUseMD5Message = SystemSecurityCryptographyResources.DoNotUseMD5;
-        private readonly string _doNotUseSHA1Message = SystemSecurityCryptographyResources.DoNotUseSHA1;
+        private const string CA5350RuleName = DoNotUseInsecureCryptographicAlgorithmsAnalyzer.DoNotUseWeakCryptographyRuleId;
+        private const string CA5351RuleName = DoNotUseInsecureCryptographicAlgorithmsAnalyzer.DoNotUseBrokenCryptographyRuleId;
+
+        private static readonly string s_CA5350RuleTitle = SystemSecurityCryptographyResources.DoNotUseWeakCryptographicAlgorithms;
+        private static readonly string s_CA5351RuleTitle = SystemSecurityCryptographyResources.DoNotUseBrokenCryptographicAlgorithms;
+
+        private static readonly string s_CA5350RuleMessage = SystemSecurityCryptographyResources.DoNotUseWeakCryptographicAlgorithmsMessage;
+        private static readonly string s_CA5351RuleMessage = SystemSecurityCryptographyResources.DoNotUseBrokenCryptographicAlgorithmsMessage;
+
+        private static readonly DiagnosticDescriptor s_CA5350Rule =
```

Comment:
Instead of creating a descriptor here, can you directly access the descriptor field on the analyzer?